server/controllers/conversation.js

```diff
@@ -0,0 +1,167 @@
+
+
+const User = require( "../models/User" );
+const asyncHandler = require( "express-async-handler" );
+const Conversation = require( "../models/conversation" );
+const Message = require( "../models/message" );
+const Profile = require( "../models/Profile" );
+
+// @route Post /create
+// @desc create conversation
+// @access Public
+exports.createConversation = asyncHandler( async ( req, res, next ) => {
+  const userId = await User.findById( req.user.id );
+  if ( !userId )
+  {
+    res.status( 401 );
+    throw new Error( "Not authorized" );
+  }
+  const senderId = req.user.id;
+  const { recipientId, text, conversationId } = req.body;
+  if ( conversationId )
+  {
+    const message = await Message.create( { senderId, text, conversationId } );
+    return res.json( { message } );
+  }
+
+  let conversation = await Conversation.findOne( { $and: [ { userId1: senderId }, { userId2: recipientId } ] } );
+  if ( conversation )
+  {
+    const message = await Message.create( {
+      senderId,
+      text,
+      conversationId: conversation._id
+    } )
+    if ( message )
+    {
+      res.status( 200 ).json( {
+        success: {
+          message: message,
+          conversation: conversation._id
+        },
+      } );
+    }
+    else
```

Comment:
I don't think we will reach this `else` clause as the asyncHandler will handle the promise rejection and forward it to the error middleware

Ditto where applicable

Reply:
removed the else clause